lib/functions/aeminitiateupload.js

```diff
@@ -104,11 +106,27 @@ class AEMInitiateUpload extends AsyncGeneratorFunction {
             const completeUrl = new URL(initiateResponse.completeURI, folderUrl);
             for (let i = 0; i < assets.length; ++i) {
                 const transferAsset = assets[i];
-                const { minPartSize, maxPartSize, uploadURIs, uploadToken } = files[i];
+                const { minPartSize, maxPartSize, uploadURIs, mimeType, uploadToken } = files[i];
+
                 // uploadToken is opaque to us, we just need to send it back with the completeUpload
                 if (!Number.isFinite(minPartSize) || !Number.isFinite(maxPartSize) || !Array.isArray(uploadURIs) || (uploadURIs.length === 0)) {
                     throw Error(`invalid multi-part information for ${transferAsset.target.url}: ${JSON.stringify(files[i])}`);
                 }
+
+                // if the client has no mimetype specified, use the mimetype provided by AEM, if AEM
+                // cannot detect a mimetype default to application/octet-stream
+                if (!transferAsset.metadata.contentType) {
+                    let contentType = mimeType;
+                    if (typeof contentType !== "string") {
+                        contentType = MIMETYPE.APPLICATION_OCTET_STREAM;
+                    }
+                    transferAsset.metadata = new AssetMetadata(
+                        transferAsset.metadata.filename,
+                        contentType,
+                        transferAsset.metadata.contentLength
+                    );
+                }              
```

Comment:
Key part of the bug fix, pass along the `mimeType` provided by AEM in to pipeline. Allows the client to provide their own mimeType, and defaults to `application/octet-stream` if no mimetype can be determined.

Reply:
Should the default `application/octet-stream` be set in `transferAsset.metadata.contentType` too? In case it's used later (with the assumption it would have been set to a sensible default if it couldn't be determined)?

Reply:
It is, the `metadata` field is replaced and after this point `contentType` is either:

- the value provided _before_ initiate upload was called
- the value provided in the initiate upload response 
- or as a fallback application/octet-stream